src/scaler/spreadscaler/provider.rs

```diff
@@ -214,14 +222,38 @@ impl<S: ReadStore + Send + Sync + Clone> Scaler for ProviderSpreadScaler<S> {
                                 })
                             })
                             .take(num_to_start)
-                            .collect::<Vec<Command>>()
+                            .collect::<Vec<Command>>();
+
+                        if commands.len() < num_to_start {
+                            let msg = format!("Could not satisfy spread {} for {}, {}/{} eligible hosts found.", spread.name, self.config.provider_reference, running_for_spread.len(), count);
```

Comment:
Curious why we compare `commands.len()` to `num_to_start`, but then report `running_for_spread.len()` out of `count`

Reply:
Yeah, I ran right into this since I reported what we compare at first. Turns out if I wanted to start 2 replicas of a provider, and one is already running, then `num_to_start` is 1 and `commands.len()` is 0. What we really want to report here though is that we already found an eligible host but something is running there, showing `1/2 eligible hosts found`, rather than `0/1`.

Reply:
Might want to add a comment here explaining the logic behind it